state.go

```diff
@@ -446,7 +446,10 @@ func (u *UpdateFetchState) Handle(ctx *StateContext, c Controller) (State, bool)
 	}
 
 	// report downloading, don't care about errors
-	c.ReportUpdateStatus(u.update, client.StatusDownloading)
+	merr := c.ReportUpdateStatus(u.update, client.StatusDownloading)
+	if merr != nil && merr.IsFatal() {
+		return NewUpdateErrorState(NewTransientError(merr.Cause()), u.update), false
```

Comment:
Ah, I see. Yes, that makes sense. The effect of the error was that you entered the error state, and now you want to produce a "softer" error to do cleanup.